packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [
+    componentTree,
+    getComponentMetadata,
+    activePage,
+    UUIDToFileMetadata,
+    pageCss,
+  ] = useStudioStore((store) => [
+    store.actions.getComponentTree(),
+    store.actions.getComponentMetadata,
+    store.pages.activePageName,
+    store.fileMetadatas.UUIDToFileMetadata,
+    store.pages.getActivePageState()?.cssImports,
+  ]);
+  useEffect(() => {
+    if (!iframeDocument) {
+      return;
+    }
+    componentTree?.forEach((component) => {
+      const cssImports = getComponentMetadata(component)?.cssImports;
+      cssImports?.forEach((cssFilepath) => {
+        injectStyleIntoIframe(iframeDocument, cssFilepath);
+      });
+    });
+    pageCss?.forEach((cssFilepath) => {
+      injectStyleIntoIframe(iframeDocument, cssFilepath);
+    });
+
+    return () => {
+      clearStylingFromIframe(iframeDocument);
+    };
+  }, [
+    componentTree,
+    getComponentMetadata,
+    iframeDocument,
+    pageCss,
+    activePage,
+    UUIDToFileMetadata,
+  ]);
+}
+
+function clearStylingFromIframe(iframeDocument: Document) {
+  const styleElements = [...iframeDocument.head.getElementsByTagName("style")];
+  styleElements.forEach((element) => {
+    element.remove();
```

Comment:
if there was studio styling for a component within the iframe, could this be a problem?

Reply:
There should never be a studio styling style tag in the iframe!  We only move styletags into the iframe that are imported by the user.  If the user and Studio happen to be using the same CSS, there would be two versions of the styletag in Studio's (parent) head (one from vite that we ignore using the `?studioStyling` query param, and one with our custom attribute, `studio-user-custom-style`. The only styletags that we care about for the iframe are the ones with `studio-user-custom-style`.

Reply:
So even if a Studio component is only used within `IFramePortal`, Vite would still add the styletags for its CSS (because it would have `?studioStyling`) to the parent document's head rather than the iframe's?